hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,202 @@ This document aims to make the distinction between physical entropy and factory
 However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
 That is either physical entropy, or the output of a DRNG which itself has been seeded (and possibly reseeded) with physical entropy.
 In CC terms, "entropy strings" (when used in this document without a qualifier) should be understood to come from either a PTG.2 or PTG.3 class RNG.
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+
+## Block Diagram
+
+
+
+## Hardware Interfaces
+
+
+
+## Design Details
+
+### Application Interface
+
+This section describes the application interface.
+The general interface operates like a FIFO, in which a command is pushed into the interface.
+An optional stream of data may follow, such as seed material for an instantiate command.
+There is one instance of a software interface.
+Many instances of the hardware interface maybe exist, and is determined by the NHwApp parameter.
+
+#### General Command Format
+
+Below is the general format for the application interface commands.
+The following fields are expanded below for clarity.
+
+- `acmd` : application command - selects one of five operations to perform
+- `shid` : state handle identification - selects which csrng maintained state to use
```

Comment:
strictly speaking are these csrng maintained states called "instances"?

Reply:
NIST refers to these states as a "working_state", and is referenced by a "state handle".  I have heard others refer to the state collection as instances, but there is no reference to state in this way by the NIST recommendation.